policy/modules/kernel/domain.te

```diff
@@ -131,6 +131,7 @@ files_list_root(domain)
 ifdef(`init_systemd',`
 	optional_policy(`
 		shutdown_sigchld(domain)
+		systemd_stream_connect_nsresourced(domain)
```

Comment:
It seems this is too broad. It would make sense to limit this to domains that have `user_namespace create`, no?

Reply:
According to my test results, many domains need to access this unix socket, not just the domains that have user_namespace create.
As said in https://systemd.io/USER_GROUP_API : Each subsystem that needs to define users and groups on the local system is supposed to implement this API, and offer its interfaces on a Varlink AF_UNIX/SOCK_STREAM file system socket bound into the /run/systemd/userdb/ directory.

type=AVC msg=audit(1724929327.084:58): avc:  denied  { write } for  pid=259 comm="avahi-daemon" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:avahi_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929327.111:59): avc:  denied  { write } for  pid=262 comm="dbus-daemon" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:system_dbusd_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929327.119:60): avc:  denied  { write } for  pid=262 comm="dbus-daemon" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:system_dbusd_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929327.131:61): avc:  denied  { write } for  pid=262 comm="dbus-daemon" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:system_dbusd_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929327.138:63): avc:  denied  { write } for  pid=262 comm="dbus-daemon" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:system_dbusd_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929327.154:66): avc:  denied  { write } for  pid=262 comm="dbus-daemon" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:system_dbusd_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929328.020:84): avc:  denied  { write } for  pid=363 comm="named" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:named_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929328.386:88): avc:  denied  { write } for  pid=447 comm="qmgr" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:postfix_qmgr_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929328.387:89): avc:  denied  { write } for  pid=446 comm="pickup" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:postfix_pickup_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929335.389:93): avc:  denied  { write } for  pid=249 comm="systemd-userwor" name="io.systemd.NamespaceResource" dev="tmpfs" ino=58 scontext=system_u:system_r:systemd_userdbd_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresourced_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929335.395:94): avc:  denied  { write } for  pid=243 comm="systemd-userwor" name="io.systemd.NamespaceResou
rce" dev="tmpfs" ino=58 scontext=system_u:system_r:systemd_userdbd_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresource
d_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929335.497:97): avc:  denied  { write } for  pid=241 comm="systemd-userwor" name="io.systemd.NamespaceResou
rce" dev="tmpfs" ino=58 scontext=system_u:system_r:systemd_userdbd_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresource
d_runtime_t:s0 tclass=sock_file permissive=0

type=AVC msg=audit(1724929335.502:98): avc:  denied  { write } for  pid=242 comm="systemd-userwor" name="io.systemd.NamespaceResou
rce" dev="tmpfs" ino=58 scontext=system_u:system_r:systemd_userdbd_t:s0-s15:c0.c1023 tcontext=system_u:object_r:systemd_nsresource
d_runtime_t:s0 tclass=sock_file permissive=0

Reply:
> According to my test results, many domains need to access this unix socket, not just the domains that have user_namespace create.
> As said in https://systemd.io/USER_GROUP_API : Each subsystem that needs to define users and groups on the local system is supposed to implement this API, and offer its interfaces on a Varlink AF_UNIX/SOCK_STREAM file system socket bound into the /run/systemd/userdb/ directory.

This still seems much less than most/all of `domain` needing access.

Reply:
OK. I will add this rule for each service that needs it.